File: resultcapture-service/src/main/java/com/devotion/capture/service/SurveyCaptureService.java
package com.devotion.capture.service;

import com.devotion.capture.dto.QuestionAnswerDto;
import com.devotion.capture.model.QuestionAnswer;
import com.devotion.capture.model.User;

import java.util.List;

public interface SurveyCaptureService {

    void submitWholeSurvey(User user, List<QuestionAnswerDto> surveyAnswers, String surveyId);

    List<QuestionAnswer> getAnswersOnQuestion(String surveyId, Integer questionId);
}

File: resultcapture-service/src/main/kotlin/com/devotion/capture/service/SurveyCaptureService.kt
package com.devotion.capture.service

import com.devotion.capture.dto.QuestionAnswerDto
import com.devotion.capture.model.AnonimousUser
import com.devotion.capture.model.QuestionAnswer

interface SurveyCaptureService {

    fun submitWholeSurvey(user: AnonimousUser, surveyAnswers: List<QuestionAnswerDto>, surveyId: String)

    fun getAnswersOnQuestion(surveyId: String, questionId: Int?): List<QuestionAnswer>
}
